include/openssl/conf.h

```diff
@@ -21,37 +21,14 @@
 extern "C" {
 #endif
 
-typedef struct {
+typedef struct conf_value_st {
     char *section;
     char *name;
     char *value;
 } CONF_VALUE;
 
 DEFINE_STACK_OF(CONF_VALUE)
 DEFINE_LHASH_OF(CONF_VALUE);
-
-struct conf_st;
-struct conf_method_st;
-typedef struct conf_method_st CONF_METHOD;
-
-struct conf_method_st {
-    const char *name;
-    CONF *(*create) (CONF_METHOD *meth);
-    int (*init) (CONF *conf);
-    int (*destroy) (CONF *conf);
-    int (*destroy_data) (CONF *conf);
-    int (*load_bio) (CONF *conf, BIO *bp, long *eline);
-    int (*dump) (const CONF *conf, BIO *bp);
-    int (*is_number) (const CONF *conf, char c);
-    int (*to_int) (const CONF *conf, char c);
-    int (*load) (CONF *conf, const char *name, long *eline);
```

Comment:
This seems to be a breaking change?

I can kind of understand the _CONF* removal since this was fairly clearly never intended for end user use - but making conf_method_st opaque is quite different.

Reply:
Yes it is making a structure opaque. @levitte is okay with that.  If the OMC is not okay with that (i.e., someone puts a hold/-1), that's fine of course; I'll close the PR and delete the branch after a few days to give someone else the chance to pick this up and take it over.

Without the structure moving to an internal file, this is just renames to remove some functions from the public API, which seems similarly wrong -- @mattcaswell 's rationale doesn't work for me :) -- and one code cleanup.  So not worth it to me.

Reply:
Actually, @mattcaswell has a point. I had not looked closely enough at CONF_METHOD earlier.

This PR actually does two things, which do not have to be done at the same time:
1. make internal functions actually internal, which I find valuable in itself
2. make CONF_METHOD opaque

You could split this PR in two, one for each if these topics

Reply:
I'm sorry, I don't see why folks believe ```_CONF_xxx``` functions, which are declared in a public header file, and have comments about function renames, are okay to remove, but a completely undocumented/uncommented structure is not okay to remove.

I won't be doing that.  I can see where this is headed -- no harm, no foul, no bad feelings -- so I'm going to close this PR.  Someone interested may want to pick up the branch in the next few days.

Reply:
_CONF_xxxx were explicitly said to be internal even though they are in public headers. The reason they weren't place in an internal header was that *at the time*, we simply didn't think in terms we do today, with the set of internal headers.

So if anything, making those internal functions truly internal is nothing other than technical debt that should have been fixed long ago. 

CONF_METHOD doesn't have the same explicit internal... What to call it, "attribute"?

Reply:
> _CONF_xxxx were explicitly said to be internal

I must have missed that; where?

Reply:
You know, it seems I misremember things. I had to go look at the entry I wrote in CHANGES to remind myself... this was ages ago.

Suffice to say that the reminder becomes my last nail in this PRs coffin. 
(that being said, the config reader could certainly use a rework, but that will have to wait)